### DISCUSSION TITLE
Save bookmarks.json file in workspace root directory (Not .vscode/ directory)

### DISCUSSION THREAD

**Discussion by trymeouteh** [alefragnani/vscode-bookmarks#741] [Ideas]
Save bookmarks.json file in workspace root directory (Not .vscode/ directory).

This way the bookmark feature can be used in other IDEs that support bookmarks.

Would suggest creating some documentation on IDE bookmark to start a IDE bookmark standard which will contain

- How bookmarks are stored in the `bookmarks.json` file in the workplace directory
- The syntax of the `bookmarks.json` file

**Comment by alefragnani** [comment]
Hi @trymeouteh ,

The bookmarks are stored inside the `.vscode` folder because that's how VS Code extensions should store their files, similar to other IDEs. Being inside the `.vscode` folder does not blocks other IDEs to load it, because it is a regular `.json` file.

Either way, there is already an open issue (#685) asking for custom location to the `bookmarks.json` file, so your idea could take advantage of this issue.

Hope this helps

**Reply by trymeouteh** [reply]
https://github.com/alefragnani/vscode-bookmarks/issues/685 is a good feature to have, however I would still suggest setting the default directory to the workspace root directory to make these bookmarks a standard for all editor/ Anything that is stored inside the `.vscode` workspace directory indicates it is only for VS Code editor and not any other editor.

**Reply by alefragnani** [reply]
Hi @trymeouteh ,

This is a VS Code extension, so the default setting will be to use the `.vscode` folder. 

If it was built as a cross editor tool, not only the location but probably its content would need to be a bit different too

Hope this helps
